cmd/juju/status/status_internal_test.go

```diff
@@ -5435,6 +5435,123 @@ foo/0  waiting   allocating  10.0.0.1  80/TCP
 `[1:])
 }
 
+func (s *StatusSuite) TestFormatTabularManyPorts(c *gc.C) {
+	fStatus := formattedStatus{
+		Model: modelStatus{
+			Type: "caas",
+		},
+		Applications: map[string]applicationStatus{
+			"foo": {
+				Scale:   1,
+				Address: "54.32.1.2",
+				Units: map[string]unitStatus{
+					"foo/0": {
+						Address:     "10.0.0.1",
+						OpenedPorts: []string{"1555/TCP", "123/UDP", "ICMP", "80/TCP"},
+						JujuStatusInfo: statusInfoContents{
+							Current: status.Allocating,
+						},
+						WorkloadStatusInfo: statusInfoContents{
+							Current: status.Waiting,
+						},
+					},
+				},
+			},
+		},
+	}
+	out := &bytes.Buffer{}
+	err := FormatTabular(out, false, fStatus)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(out.String(), gc.Equals, `
+Model  Controller  Cloud/Region  Version
+                                 
+
+App  Version  Status  Scale  Charm  Channel  Rev  Address    Exposed  Message
+foo                     0/1                    0  54.32.1.2  no       
+
+Unit   Workload  Agent       Address   Ports                            Message
+foo/0  waiting   allocating  10.0.0.1  1555/TCP, 123/UDP, ICMP, 80/TCP  
+`[1:])
+}
+
+func (s *StatusSuite) TestFormatTabularManyPortsGrouped(c *gc.C) {
+	fStatus := formattedStatus{
+		Model: modelStatus{
+			Type: "caas",
+		},
+		Applications: map[string]applicationStatus{
+			"foo": {
+				Scale:   1,
+				Address: "54.32.1.2",
+				Units: map[string]unitStatus{
+					"foo/0": {
+						Address:     "10.0.0.1",
+						OpenedPorts: []string{"1557/TCP", "1555/TCP", "80/TCP", "ICMP", "1556/TCP"},
+						JujuStatusInfo: statusInfoContents{
+							Current: status.Allocating,
+						},
+						WorkloadStatusInfo: statusInfoContents{
+							Current: status.Waiting,
+						},
+					},
+				},
+			},
+		},
+	}
+	out := &bytes.Buffer{}
+	err := FormatTabular(out, false, fStatus)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(out.String(), gc.Equals, `
+Model  Controller  Cloud/Region  Version
+                                 
+
+App  Version  Status  Scale  Charm  Channel  Rev  Address    Exposed  Message
+foo                     0/1                    0  54.32.1.2  no       
+
+Unit   Workload  Agent       Address   Ports                        Message
+foo/0  waiting   allocating  10.0.0.1  1555-1557/TCP, 80/TCP, ICMP  
```

Comment:
Did we want space instead of comma separator, this could be
`1555-1557,80/TCP IMCP`
and in general
`1555/TCP 123/UDP ICMP 80/TCP`

Not sure if that's better? Allows for more grouping?

And then should all the same protocol go together, sorted?

`80,666-669,1555/TCP 123/UDP ICMP`

Reply:
I'd like to keep the churn to 3.0 client.

Spaces instead of commas works and I've made that change.

I didn't sort them because the old way didn't sort them, but can do this if you would prefer it.